qcodes/plots/qcmatplotlib.py

```diff
@@ -457,9 +457,11 @@ def _draw_pcolormesh(self, ax, z, x=None, y=None, subplot=1,
         pc = ax.pcolormesh(*args, **kwargs)
 
         # Scale colors from clim kwarg, or otherwise from min(z) and max(z)
-        data_arrays = [data_array.get_array() for data_array in ax.collections]
+        data_arrays = [data_array.get_array() for data_array in ax.collections
+                       if data_array.get_array() is not None]
```

Comment:
At the risk of making the code slightly more clunky, wrapping the function call in a generator ensures you only call get_array once, rather than twice saving some precious clock cycles. This would really only be noticeable if you had perhaps thousands of arrays all plotted on a single axis. 

```suggestion
        data_arrays = [arr for arr in (data_array.get_array() for data_array in ax.collections)
                       if arr is not None]
```

Reply:
I might ignore this one because while it does save an extra function call per array, the code is harder to comprehend. It's not really a big overhead in this situation